### PR TITLE
Improve 1Password integration and docs, bump connect-sdk-go to 1.5.0

### PR DESCRIPTION
**Pull request by snarlysodboxer**
This implements a newer version of 1Password's connect-sdk-go with some bug fixes. It also slightly simplifies the code.

I believe this should also fix https://github.com/external-secrets/external-secrets/issues/1202.

### CODE REVIEW

**Comment by snarlysodboxer** [comment]
Turns out this needs to wait on https://github.com/1Password/connect/issues/42 getting fixed. Sorry for the premature submission. Do you want me to close this for now? The code here won't need changed once it's fixed, other than to specify v1.5.6 of the Connect Server in the docs, when that comes out.

**Comment by gusfcarvalho** [comment]
Hey @snarlysodboxer ! I don't think so. Maybe we should only update the docs to say it needs connect v1.5.6  and thats' it

File: docs/provider-1password-automation.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ _**The 1Password API calls the entries in vaults 'Items'. These docs use the sam
     * External Secrets does not run this server. See [Deploy a Connect Server](#deploy-a-connect-server).
     * One Connect Server is needed per 1Password Automation Environment.
     * Many Vaults can be added to an Automation Environment, and Tokens can be generated in that Environment with access to any set or subset of those Vaults.
-* 1Password Connect Server version 1.3.0 or higher. 1.3.0 and 1.5.0 have been tested.
+* 1Password Connect Server version 1.5.5 or higher.
```
Comment (gusfcarvalho) [inline docs/provider-1password-automation.md:35]:
```suggestion
* 1Password Connect Server version 1.5.6 or higher.
```

Comment (snarlysodboxer) [reply docs/provider-1password-automation.md]: Done.

**Comment by gusfcarvalho** [comment]
I cannot really test vault integration with my setup, so I'm considering that things would run smoothly here :smile:

**Comment by gusfcarvalho** [comment]
thanks for the update!

**Pull request merged**